src/test/java/org/sonar/plugins/stash/StashIssueReportingPostJobTest.java

```diff
@@ -247,6 +252,36 @@ public void testExecuteOnWithNoDiffReport() throws Exception {
     verify(stashRequestFacade, times(0)).resetPullRequestApproval(eq(pr), (StashClient)Mockito.anyObject());
   }
 
+  @Test
+  public void testShouldApprovePullRequest() {
+    Issue minorIssue = new DefaultIssue().setSeverity(Severity.MINOR);
+    Issue majorIssue = new DefaultIssue().setSeverity(Severity.MAJOR);
+
+    report = new ArrayList<>();
+
+    report.add(minorIssue);
+    report.add(majorIssue);
+
+    assertFalse(
+        StashIssueReportingPostJob.shouldApprovePullRequest(Optional.empty(), report)
+    );
+
+    report.clear();
+    assertTrue(
+        StashIssueReportingPostJob.shouldApprovePullRequest(Optional.empty(), report)
+    );
+
+    report.add(minorIssue);
+    assertTrue(
+        StashIssueReportingPostJob.shouldApprovePullRequest(Optional.of(Severity.MINOR), report)

```

Comment:
As far as I can see, shouldApprovePullRequest is an internal method, that is implementation details.
It is ok to test it until one wants to change internal implementation details. Normally tests should not be modified upon such a change to ensure public contract is not changed. However when internal methods are exposed and tested, tests must be modified too. It can lead to extra work